selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java

```diff
@@ -1296,32 +1330,64 @@ public void openJavaDocPopUp() {
    * @param text is the text of elements active line
    */
   public void waitTextElementsActiveLine(final String text) {
-    waitActiveEditor();
+    waitActive();
     redrawDriverWait.until(visibilityOf(activeLineXpath));
     redrawDriverWait.until(
         (ExpectedCondition<Boolean>) driver -> activeLineXpath.getText().contains(text));
   }
 
   /** get positions of the current line and char */
-  public Pair<Integer, Integer> getCurrentCursorPositions() {
-    waitActiveEditor();
+  public Pair<Integer, Integer> getCurrentCursorPositionsFromVisible() {
+    waitActive();
     WebElement currentActiveElement =
         activeLineNumbers.stream().filter(webElement -> webElement.isDisplayed()).findFirst().get();
-    int[] currentCursorPosition =
-        Arrays.asList(
-                redrawDriverWait.until(visibilityOf(currentActiveElement)).getText().split(":"))
+    return getCursorPositionFromWebElement(currentActiveElement);
+  }
+
+  /**
+   * Check that editor active and focused on first step. After that get cursor position. Note! For
+   * correct work a editor must be active.
+   *
+   * @return char and line position from current visible, active and focused editor (usual uses with
+   *     split editor)
+   */
+  public Pair<Integer, Integer> getCursorPositionsFromCurentActive() {
+    String xpathToCurrentActiveCursorPosition =
+        "//div[@active and @focused]/parent::div[@id='gwt-debug-multiSplitPanel-tabsPanel']/div[@active and @focused]/parent::div[@id='gwt-debug-multiSplitPanel-tabsPanel']/parent::div/parent::div/following-sibling::div//div[@active]//div[@id='gwt-debug-cursorPosition']";
+    waitActive();
+    return getCursorPositionFromWebElement(
+        redrawDriverWait.until(
+            ExpectedConditions.visibilityOfElementLocated(
+                By.xpath(xpathToCurrentActiveCursorPosition))));
+  }
+
+  /**
+   * wait specified values for Line and Char cursor positions in the Codenvy editor
+   *
+   * @param linePosition expected line position
+   * @param charPosition expected char position
+   */
+  public void waitLineAndCharForCurrentActive(final int linePosition, final int charPosition) {
```

Comment:
IMHO, it's better to rename the method to have a shorter name **waitCursorInPosition()** .